src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -629,7 +629,7 @@ namespace smt::noodler {
             Concat side = occurrs.begin()->position > 0 ? act_pred.get_right_side() : act_pred.get_left_side();
 
             int start = std::abs(occurrs.begin()->position) - 1;
-            for(int i = start; i < side.size(); i++) {
+            for(int i = start; i < (int)side.size(); i++) {
```

Comment:
Why not
```suggestion
            for (size_t i = start; i < side.size(); i++) {
```
?
Does `side.size()` return signed?

Reply:
I was not sure if `start` might be negative or not. `side.size()` should be `size_t` imo.

Reply:
Ah, I see it now. Hmm, but then the code inside the loop could be buggy, we are doing `side[i]` a lot.

Do you remember why `start` is initialized the way it is initalized? Why is there `std::abs(occurrs.begin()->position)`? Can `occurrs.begin()->position` be even negative?

Reply:
I have checked it. It is supposed to be positive. Fixed it.